src/test/FuzzerImpl.cpp

```diff
@@ -503,17 +517,82 @@ isBadOverlayFuzzerInput(StellarMessage const& m)
 
 static void
 attemptToApplyOps(LedgerTxn& ltx, PublicKey const& sourceAccount,
-                  xdr::xvector<Operation> const& ops, Application& app)
+                  xdr::xvector<Operation> const& ops, Application& app,
+                  bool const throwIfTxFails = true)
 {
     auto txFramePtr =
         createFuzzTransactionFrame(sourceAccount, ops, app.getNetworkID());
     txFramePtr->attemptApplication(app, ltx);
 
-    if (txFramePtr->getResultCode() != txSUCCESS)
+    if (throwIfTxFails)
     {
-        throw std::runtime_error(fmt::format(
-            FMT_STRING("Error {} while applying operations while fuzzing"),
-            txFramePtr->getResultCode()));
+        auto const opFrames = txFramePtr->getOperations();
+        if (txFramePtr->getResultCode() != txSUCCESS)
+        {
+            auto const msg = fmt::format(
+                FMT_STRING("Error {} while applying operations while fuzzing"),
+                txFramePtr->getResultCode());
+            LOG_FATAL(DEFAULT_LOG, "{}", msg);
+            for (auto const& op : opFrames)
+            {
+                LOG_ERROR(DEFAULT_LOG, "op {}: result code {}",
+                          op->getOperation().body._xdr_discriminant(),
+                          op->getResultCode());
+            }
+            throw std::runtime_error(msg);
+        }
+        for (auto const& op : opFrames)
+        {
+            if ((op->getOperation().body.type() == MANAGE_BUY_OFFER &&
+                 op->getResult()
+                         .tr()
+                         .manageBuyOfferResult()
+                         .success()
+                         .offer.effect() == MANAGE_OFFER_DELETED) ||
+                (op->getOperation().body.type() == MANAGE_SELL_OFFER &&
+                 op->getResult()
+                         .tr()
+                         .manageSellOfferResult()
+                         .success()
+                         .offer.effect() == MANAGE_OFFER_DELETED))
+            {
+                auto const msg =
+                    fmt::format(FMT_STRING("MANAGE_OFFER_DELETED while setting "
+                                           "up fuzzing -- op type {}"),
+                                op->getOperation().body.type());
+                LOG_FATAL(DEFAULT_LOG, "{}", msg);
+                if (op->getOperation().body.type() == MANAGE_SELL_OFFER)
+                {
+                    LOG_FATAL(
```

Comment:
A few comments here:
- This would have been way nicer if you had taken a reference to `op->getOperation().body.manageSellOfferOp()` and use it in the logging. Analogous comment applies on 581,591.
- This code is essentially a duplicate of the code on 581,591. You can easily refactor this into a static (this compilation unit only) template function. You could also use the lambda auto argument functionality to achieve the same.

Reply:
But more importantly, why even write this log message? You could do something like
```
auto const msg = fmt::format(FMT_STRING("MANAGE_OFFER_DELETED while setting up fuzzing. Operation is {}"), xdr_to_string(op));
```
Always go for the simple solution. Try to use as much existing functionality as possible. If you see yourself writing a lot of code that you've never seen anywhere else in stellar-core, you should probably be suspicious that there is an easier way.

Reply:
> If you see yourself writing a lot of code that you've never seen anywhere else in stellar-core, you should probably be suspicious that there is an easier way.

Well put.

I've cleaned up the logging in `attemptToApplyOps()`.